main.tf

```diff
@@ -14,7 +14,83 @@
  * limitations under the License.
  */
 
-resource "google_storage_bucket" "main" {
-  project = var.project_id
-  name    = var.bucket_name
+data "google_organization" "org" {
+  count  = var.domain != "" ? 1 : 0
+  domain = var.domain
+}
+
+locals {
+  customer_id = var.domain != "" ? data.google_organization.org[0].directory_customer_id : var.customer_id
+  type        = "default"
+  label_keys = {
+    "default" = "cloudidentity.googleapis.com/groups.discussion_forum"
+    # Placeholders according to https://cloud.google.com/identity/docs/groups#group_properties.
+    # Not supported by provider yet.
+    "dynamic"  = "cloudidentity.googleapis.com/groups.dynamic"
+    "security" = "cloudidentity.googleapis.com/groups.security"
+    "external" = "system/groups/external"
+  }
+}
+
+resource "null_resource" "customer_id_is_empty" {
```

Comment:
Instead of the null_resource wont this be automatically surfaced by the API when `parent = "customers/"` as it is invalid?

Reply:
Removed.

That's right. I was sure which is better, either error out at module level or API level. Either way is fine with me.